onnxruntime/core/graph/model_load_utils.h

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <list>
+#include <unordered_map>
+#include <memory>
+#include <string>
+#include "gsl/gsl"
+#include "core/platform/env.h"
+#include "core/common/common.h"
+
+namespace onnxruntime {
+
+namespace model_load_utils {
+
+static constexpr const char* kAllowReleasedONNXOpsetOnly = "ALLOW_RELEASED_ONNX_OPSET_ONLY";
+
+inline bool ISAllowReleasedONNXOpsetsOnlySet() {
+  // Get the value of env variable kAllowReleasedONNXOpsetOnly
+  const std::string allow_official_onnx_release_only_str =
+      Env::Default().GetEnvironmentVar(model_load_utils::kAllowReleasedONNXOpsetOnly);
+
+  if (!allow_official_onnx_release_only_str.empty()) {
+    // Check if the env var contains an unsupported value
+    if (allow_official_onnx_release_only_str.length() > 1 ||
+        (allow_official_onnx_release_only_str[0] != '0' && allow_official_onnx_release_only_str[0] != '1')) {
+      ORT_THROW("The only supported values for the environment variable ",
+                model_load_utils::kAllowReleasedONNXOpsetOnly,
+                " are '0' and '1'. The environment variable contained the value: ",
+                allow_official_onnx_release_only_str);
+    }
+
+    if (allow_official_onnx_release_only_str[0] == '0') {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+inline void ValidateOpsetForDomain(const std::unordered_map<std::string, int>& onnx_released_versions, const logging::Logger& logger,
+                                   bool allow_official_onnx_release_only,
+                                   const std::string& domain, int version) {
+  auto it = onnx_released_versions.find(domain);
+  if (it != onnx_released_versions.end() && version > it->second) {
```

Comment:
No else block for this?

Reply:
not needed because else block is valid case.